### PR TITLE
Fixed Momentum Optimization Equation

### PR DESCRIPTION
**Pull request by eduardo-elizondo**

### CODE REVIEW

**Comment by vrv** [comment]
If I remember correctly, the existing code is actually a correct reparameterization of momentum, even though it looks subtly different.

The momentum vector stores the unscaled gradients, and we multiply the entire momentum vector by the learning rate when applying the update.

(This has come up several times internally -- we should add documentation to the ApplyMomentum op: https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/ops/training_ops.cc#L98  Feel free to do so there if you think it would help others).
